src/app/pages/common/ix-forms/services/ix-formatter.service.ts

```diff
@@ -0,0 +1,227 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class IxFormatterService {
+  readonly IecUnits = ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB'];
+
+  /**
+   * Formats any memory size human readable string into a normalized size reading, e.g., '2m' to '2 MiB'
+   * @param value The string to be formatted
+   * @returns Formatted string
+   */
+  memorySizeFormatting: (val: string | number) => string = (value: string | number) => {
+    value = value.toString();
+    if (!value) {
+      return '';
+    }
+    let formatted = '';
+    if (value.replace(/\s/g, '').match(/[^0-9]/g) === null) {
+      formatted = this.convertBytestoHumanReadable(value.replace(/\s/g, ''), 0);
+    } else {
+      formatted = this.reformatHumanString(value);
```

Comment:
Why do this tho, if we always assume incoming value is a correct numeric representation?

Reply:
The first check is if the passed in string is a huge number of raw bytes and has no unit alongside it. For example, `2147483648` which should formatted to `2 GiB`. The second check is for when the value passed has deformed units or wrong units. For example, `2048 MiB` should be `2 GiB` or `2mb` should be `2 MiB`

Reply:
Same as above. This function should not receive `2048 MiB` or `2Mb` or anything else than `2147483648`